src/ray/gcs/gcs_server/gcs_actor_manager.cc

```diff
@@ -858,30 +858,30 @@ void GcsActorManager::OnActorCreationSuccess(const std::shared_ptr<GcsActor> &ac
   }
   actor->UpdateState(rpc::ActorTableData::ALIVE);
   auto actor_table_data = actor->GetActorTableData();
+  // Invoke all callbacks for all registration requests of this actor (duplicated
+  // requests are included) and remove all of them from
+  // actor_to_create_callbacks_.
+  auto iter = actor_to_create_callbacks_.find(actor_id);
+  if (iter != actor_to_create_callbacks_.end()) {
+    for (auto &callback : iter->second) {
+      callback(actor);
```

Comment:
We must write to the gcs storage before calling callback, otherwise if GCS is crash before writing to storage after calling callback, GCS client will not resend the request.

Reply:
That's a great point. I moved this block into the callback. Lmk if you can find any issue from adding actor entry to the `created_actor` before callback.